frontend/test/pytest/test_jax_integration.py

```diff
@@ -325,6 +326,27 @@ def cost_fn(x, y):
         assert "0" in circuit.jaxed_qfunc.deriv_qfuncs
         assert len(circuit.jaxed_qfunc.deriv_qfuncs["0"].jaxpr.out_avals) == 1
 
+    def test_jit_and_grad(self, backend):
+        """Test that argnum determination works correctly when combining jax.jit with jax.grad.
+        This was fixed by the introduction of symbolic zero detection for tangent vectors."""
```

Comment:
@dime10 a suggestion: to use the `monkeypatch` fixture here to ensure that the internal argnums _are_ being set correctly :)

Reply:
This has been implemented in the `test_argnums_passed` test in this file.